### PR TITLE
Split user

### PR DESCRIPTION
**Pull request by vjik**
| Q             | A
| ------------- | ---
| Is bugfix?    | ❌
| New feature?  | ❌
| Breaks BC?    | ✔️
| Fixed issues  | #19 

Adopt PRs:

| Package             | PR
| ------------- | ---
| yiisoft/app-api | https://github.com/yiisoft/app-api/pull/45
| yiisoft/yii-demo | https://github.com/yiisoft/yii-demo/pull/244
| yiisoft/demo-api | https://github.com/yiisoft/demo-api/pull/6

### CODE REVIEW

**Review by samdark — APPROVED** [review]

File: src/AuthenticatorInterface.php [hunk 1]
```diff
@@ -0,0 +1,37 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Yiisoft\User;
+
+use Yiisoft\Auth\IdentityInterface;
+
+interface AuthenticatorInterface
```
Comment (samdark) [inline src/AuthenticatorInterface.php:9]: It doesn't authentificate so the name should be another one.

Comment (samdark) [reply src/AuthenticatorInterface.php:9]: - CurrentIdentity
  - get()
  - set(IdentityInterface $identity)
  - clear()

File: composer.json [hunk 2]
```diff
@@ -23,7 +23,6 @@
         "psr/event-dispatcher": "^1.0",
         "psr/http-factory": "^1.0",
         "psr/log": "^1.1",
-        "yiisoft/access": "^1.0",
```
Comment (samdark) [inline composer.json:26]: Previously you were able to do

```php
if ($currentUser->can('update_post', $post)) {
   $post->setModifiedBy($currentUser);
   $postRepository->persist($post);
}
```

How to do it now?

File: src/CurrentIdentity/Storage/MemoryCurrentIdentityStorage.php [hunk 3]
```diff
@@ -0,0 +1,28 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Yiisoft\User\CurrentIdentity\Storage;
+
+/**
+ * Stores current identity ID in memory.
+ */
+final class MemoryCurrentIdentityStorage implements CurrentIdentityStorageInterface
```
Comment (samdark) [inline src/CurrentIdentity/Storage/MemoryCurrentIdentityStorage.php:10]: Is it used anywhere?

File: src/CurrentIdentity/Storage/SessionCurrentIdentityStorage.php [hunk 4]
```diff
@@ -0,0 +1,125 @@
...
+    private ?int $absoluteAuthTimeout = null;
+
+    private SessionInterface $session;
+
+    public function __construct(SessionInterface $session)
+    {
+        $this->session = $session;
+    }
+
+    public function setAuthTimeout(int $timeout = null): self
```
Comment (samdark) [inline src/CurrentIdentity/Storage/SessionCurrentIdentityStorage.php:38]: It's not part of the interface. Should it be supported by other storages and be part of the interface or not?

File: composer.json (hunk 2 above)
Comment (vjik) [reply composer.json]: Returned the `can` method.

File: src/CurrentIdentity/Storage/MemoryCurrentIdentityStorage.php (hunk 3 above)
Comment (vjik) [reply src/CurrentIdentity/Storage/MemoryCurrentIdentityStorage.php:10]: Authentication has to be performed on each request, which is often the case for stateless application such as RESTful API.

File: src/CurrentIdentity/Storage/SessionCurrentIdentityStorage.php (hunk 4 above)
Comment (vjik) [reply src/CurrentIdentity/Storage/SessionCurrentIdentityStorage.php:38]: This method not need in other places. I think use this method only for concrete implementation.

**Comment by samdark** [comment]
Decided on a call that we need to eliminate storage interface and use session only. If session is null, do not write ID.

**Review by samdark — APPROVED** [review]

**Comment by samdark** [comment]
👍

**Pull request merged**